Update README with Canary info (#16125)
microsoft/terminal: README.md

---

## Installing Windows Terminal Canary

> Windows Terminal Canary is a canary build of Windows Terminal. This canary build ships nightly with the latest code from our `main` branch before they make it to Windows Terminal Preview. Windows Terminal Canary is also our least stable offering, so you may discover bugs before we have had a chance to find them.

Reviewer: ```suggestion
Windows Terminal Canary is a nightly build of Windows Terminal. This build the latest code from our `main` branch, giving you an opportunity to try features before they make it to Windows Terminal Preview.

Windows Terminal Canary is our least stable offering, so you may discover bugs before we have had a chance to find them.
```

> Windows Terminal Canary is available as a App Installer distribution and a Portable ZIP distribution. 

Reviewer: ```suggestion
Windows Terminal Canary is available as an App Installer distribution and a Portable ZIP distribution. 
```

The App Installer distribution supports automatic updates. Due to platform limitations, this installer only works on Windows 11. 

> The Portable ZIP distribution is a portable application. It will not automatically update and or automatically check for updates. This portable ZIP distribution works on Windows 10 (19041+) and Windows 11.

Reviewer: I'd add the "For more information . . . (link to distributions doc)" line under these if you can

---

## Installing Windows Terminal Canary

> Windows Terminal Canary is a nightly build of Windows Terminal. This build the latest code from our `main` branch, giving you an opportunity to try features before they make it to Windows Terminal Preview.

Reviewer: ```suggestion
Windows Terminal Canary is a nightly build of Windows Terminal. This build has the latest code from our `main` branch, giving you an opportunity to try features before they make it to Windows Terminal Preview.
```

Reviewer: sorry, i messed that one up
